### DISCUSSION TITLE
How to get all failed jobs using C# SDK

### DISCUSSION THREAD

**Discussion by VenkateshSrini** [kubernetes-client/csharp#871] [Q&A]
Hi all,
I need a way to get all failed jobs in a namespace using C$ sdk. It should not get the running the jobs. It should get only failed jobs. Please help.

Regards
Venkatesh

**Comment by tg123** [comment]
first figure out how to do it with `kubectl`
one solution is `kubectl get job  | grep xxx`

thus, you should list jobs then .where them

**Comment by brendandburns** [comment]
There is a small example here:
https://github.com/kubernetes-client/csharp/blob/master/examples/simple/PodList.cs

Which shows how to list Pods. You will need to modify it so that it lists Jobs instead, and then instead of just printing every Job, you'll need to filter that list on the job status.

I'm going to close this issue, since this is more of a request for help programming something rather than an actual problem with the client library. If you run into a problem with the client library please file a new issue.

**Comment by VenkateshSrini** [comment]
@brendandburns ,
That is providing just the pod name. I tried something like this 
```
                var jobs = await _k8sClient.ListNamespacedJobAsync(nameSpace);
                var results = jobs?.Items.Select(V1Job => new ResJobDetails
                {
                    JobName = V1Job.Metadata.Name,
                    
                    JobType = V1Job.Metadata.Labels.ContainsKey("jobtype") ?
                              V1Job.Metadata.Labels["jobtype"] : string.Empty,
                    JobUid = V1Job.Metadata.Uid,
                    NameSpace = nameSpace
                }).ToList();
                foreach (var job in jobs.Items)
                {
                    var resultJob = results.FirstOrDefault(resJob => 
                                    resJob.JobName.CompareTo(job.Metadata.Name) == 0);
                    if (job.Status.Failed >0)
                    {
                        resultJob.JobStatus = "Failed";
                    }
                    else if (job.Status.Active>0)
                    {
                        resultJob.JobStatus = "Active";
                    }
                    else if (job.Status.Succeeded>0) resultJob.JobStatus = "Success";

                }
```
Can you please let e know if this correct or could there be some wrong results
